File: backend/src/main/java/eliasyazdani/capstone/cupwithme/backend/tournament/TournamentWithoutID.java
package eliasyazdani.capstone.cupwithme.backend.tournament;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Positive;

import java.util.List;

public record TournamentWithoutID(
        String admin,
        @NotBlank
        String tournamentName,
        @NotBlank
        String location,
        @Positive
        int numberOfPlayers,
        List<MatchWithoutId> matchesWithoutId,
        String champion
) {
}

File: frontend/src/Models/TournamentWithoutId.ts

import {RoundWithoutId} from "./RoundWithoutId.ts";


export type TournamentWithoutId = {
    admin: string,
    tournamentName: string,
    location: string,
    numberOfPlayers: number,
    roundsWithoutId: RoundWithoutId[],
    champion: string

}
